test/jdk/com/sun/java/swing/plaf/DesktopPropertyResetPendingFlagTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8367376
+ * @summary DesktopProperty never reset pending status to process new updates
+ * @modules java.desktop/sun.swing.plaf
+ * @run main DesktopPropertyResetPendingFlagTest
+ */
+
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.plaf.basic.BasicButtonUI;
+
+import sun.swing.plaf.DesktopProperty;
+
+import java.awt.AWTEvent;
+import java.awt.EventQueue;
+import java.awt.Toolkit;
+import java.util.concurrent.CountDownLatch;
+
+public class DesktopPropertyResetPendingFlagTest extends JFrame {
+
+    static class ExpectedException extends RuntimeException {}
+
+    /**
+     * The original ticket required changing the system accessibility settings.
+     * But we can instead automate this test if we just create & update our own
+     * DesktopProperty.
+     */
+    static class TestDesktopProperty extends DesktopProperty {
+
+        public TestDesktopProperty(String key, Object fallback) {
+            super(key, fallback);
+        }
+
+        @Override
+        public void updateUI() {
+            super.updateUI();
+        }
+    }
+
+    private static void setLookAndFeel() {
+        try {
+            String lf = UIManager.getSystemLookAndFeelClassName();
+            UIManager.setLookAndFeel(lf);
```

Comment:
Guess we should iterate for all L&F s, not just test the default..

Reply:
If you do not want to tinker with the current test, maybe you can have another test testing the L&F s,  handling exception gracefully in updateUI

Reply:
I agree with Alexey. I think we should definitely iterate across all L&Fs and it can be done without creating more tests.

Reply:
I updated the test to iterate over all available L&F's